src/librustc/infer/mod.rs

```diff
@@ -80,6 +80,16 @@ pub type Bound<T> = Option<T>;
 pub type UnitResult<'tcx> = RelateResult<'tcx, ()>; // "unify result"
 pub type FixupResult<T> = Result<T, FixupError>; // "fixup result"
 
+/// A flag that is given when running region resolution: if true, it
+/// indicates that we should not report the region errors to the user
+/// if NLL is enabled, since NLL will also detect them (and do a
+/// better job of it).
```

Comment:
1. Should this comment include the special role of `borrowck=migrate`?

2. Instead of checking for the borrowck=mode after seeing `UnlessNll(true)`, *maybe* we should revise how `UnlessNll` itself is initialized based on inspecting that `BorrowckMode`?
  * Its up to you; I just know that I initially found the name `UnlessNll` confusing, given that we *will* emit the errors under the migration mode... But of course that was inherited from the previous code...

Reply:
I feel like if we do the checking *before* initializing `UnlessNll`, I would just call it `Ignore` or something like that, since at that point it is encoding whether to ignore the error and nothing else, right?

(Right now it's saying "is this the sort of error we should ignore if NLL is enabled" — maybe I can improve the comment this way?)

Reply:
I'm not inclined to move the logic earlier because it's created many places -- well, I guess I can just have a constructor like `RegionErrors::suppress_if_nll()` or something?

Reply:
ok, did that, and it seems nice. will push a commit.